buildchain/buildchain/image.py

```diff
@@ -204,7 +207,9 @@ def _operator_image(name: str, **kwargs: Any) -> targets.OperatorImage:
     'nginx-ingress-defaultbackend-amd64': 'defaultbackend-amd64',
 }
 
-SAVE_AS_TAR : FrozenSet[str] = frozenset(('nginx', 'pause'))
+SAVE_AS_TAR : FrozenSet[str] = frozenset(['pause'])
+
+SAVE_AS_TAR_AND_IMAGE : FrozenSet[str] = frozenset(['nginx'])
```

Comment:
I'm not very fond of this approach.

It's really ad-hoc/hackish, you can see it from the name `SAVE_AS_TAR_AND_IMAGE`.
Like if we ever add a third format, we will have `SAVE_AS_TAR_AND_IMAGE_AND_FOO` (unlikely I know, but it's still a code smell ;))

This aside, the current implementation breaks the cleanup: a `doit clean` on `_image_pull:nginx` won't delete the tarball :/

I propose getting rid of the Boolean to use a list of formats instead. As a nice side effect it allows to have a simpler code in `RemoteImage` by moving the variant parts into dedicated class.
You can cherry-pick https://github.com/scality/metalk8s/commit/ac15e324be8b9deb186bfa805b0cdb841e6c5dbd if you want (should apply on top of your PR without conflict).

Reply:
Done